src/main/java/org/semver4j/internal/range/processor/RangesUtils.java

```diff
@@ -1,31 +1,36 @@
 package org.semver4j.internal.range.processor;
 
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
 import static java.lang.Integer.parseInt;
 
 /**
  * Set of methods which helps ranges handling.
  */
 final class RangesUtils {
+    @NotNull
     static final String EMPTY = "";
+    @NotNull
     static final String SPACE = " ";
 
     private static final int X_RANGE_MARKER = -1;
 
     private RangesUtils() {
     }
 
-    static int parseIntWithXSupport(String id) {
+    static int parseIntWithXSupport(@Nullable final String id) {
         if (id == null || id.equalsIgnoreCase("x") || id.equals("*") || id.equals("+")) {
             return X_RANGE_MARKER;
         }
         return parseInt(id);
     }
 
-    static boolean isX(Integer id) {
+    static boolean isX(int id) {
```

Comment:
Why you changed contract of this method? Are you sure it is correct?

Reply:
The following line is a comparation who does an unboxing, which will thrown NPE if `id` is null.
Since `id` must be not null, we do not need to use the `Integer` wrapper, which is faster.